contracts/claimManager.sol

```diff
@@ -0,0 +1,242 @@
+        require(deposits[claimant] >= minDeposit);
+
+        ScryptClaim storage claim = claims[numClaims];
+        claim.claimant = claimant;
+        claim.plaintext = _plaintext;
+        claim.blockHash = _blockHash;
+        claim.numChallengers = 0;
+        claim.currentChallenger = 0;
+        claim.verificationOngoing = false;
+        claim.createdAt = block.number;
+
+        bondDeposit(numClaims, claimant, minDeposit);
+        ClaimCreated(numClaims, claim.claimant, claim.plaintext, claim.blockHash);
+
+        numClaims.add(1);
+        return numClaims;
+    }
+
+    // @dev – challenge an existing Scrypt claim.
+    // triggers a downstream claim computation on the scryptVerifier contract
+    // where the claimant & the challenger will immediately begin playing a verification.
+    //
+    // @param claimID – the claim ID.
+    function challengeClaim(uint claimID) public {
+        ScryptClaim storage claim = claims[claimID];
+
+        // check that the claim is in Stage 1 (i.e. accepting new challenges).
+        require(block.number.sub(claim.createdAt) <= challengeTimeout);
+
+        require(deposits[msg.sender] >= minDeposit);
+        bondDeposit(claimID, msg.sender, minDeposit);
+
+        claim.challengers[claim.numChallengers] = msg.sender;
+        claim.numChallengers = claim.numChallengers.add(1);
+        ClaimChallenged(claimID, msg.sender);
+    }
+
+    // @dev – runs a verification game between the claimant and
+    // the next queued-up challenger.
+    //
+    // @param claimID – the claim id.
+    function runNextVerificationGame(uint claimID) public {
+        ScryptClaim storage claim = claims[claimID];
+
+        // check if there is a challenger who has not the played verificationg game yet.
+        if (claim.numChallengers > claim.currentChallenger) {
+            require(claim.verificationOngoing == false);
+
+            // kick off a verification game.
+            uint sessionId = sv.claimComputation(claim.challengers[claim.currentChallenger], claim.claimant, claim.plaintext, claim.blockHash, 2050);
+            ClaimVerificationGameStarted(claimID, claim.claimant, claim.challengers[claim.currentChallenger], sessionId);
+
+            claim.verificationOngoing = true;
+            claim.currentChallenger = claim.currentChallenger.add(1);
+        } else {
+            require(claim.verificationOngoing == false);
+            ClaimVerificationGamesEnded(claimID);
+        }
+    }
+
+    // @dev – called when a verification game has ended.
+    // only callable by the scryptVerifier contract.
+    //
+    // @param claimID – the claim ID.
+    // @param winner – winner of the verification game.
+    // @param loser – loser of the verification game.
+    function claimDecided(uint claimID, address winner, address loser) onlyBy(address(sv)) public {
+        ScryptClaim storage claim = claims[claimID];
+
+        claim.verificationOngoing = false;
```

Comment:
We should check that it was true before.

Reply:
Implemented in a6b0019b3b7b40dbea247aff0cbb0a0b2c6913d4.